Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBOAuthTokenRequest.m

```diff
@@ -0,0 +1,180 @@
+///
+/// Copyright (c) 2020 Dropbox, Inc. All rights reserved.
+///
+
+#import "DBOAuthTokenRequest.h"
+
+#import "DBOAuthManager.h"
+#import "DBOAuthResult.h"
+#import "DBSDKConstants.h"
+#import "DBTransportBaseConfig.h"
+#import "DBTransportBaseHostnameConfig.h"
+
+@interface DBOAuthTokenExchangeRequest ()
+
+@property (nonatomic, readwrite, strong) NSURLSessionDataTask *task;
+@property (nonatomic, readwrite, strong) DBOAuthTokenExchangeRequest* retainSelf;
+@property (nonatomic, readwrite, copy) DBOAuthCompletion completion;
+@property (nonatomic, readwrite, strong) dispatch_queue_t queue;
+
+@end
+
+@implementation DBOAuthTokenExchangeRequest
+
+- (instancetype)initWithOAuthCode:(NSString *)oauthCode
+                     codeVerifier:(NSString *)codeVerifier
+                           appKey:(NSString *)appKey
+                           locale:(NSString *)locale
+                      redirectUri:(NSString *)redirectUri {
+  self = [super init];
+  if (self) {
+    _task = [self db_createTokenRequestTaskWithOAuthCode:oauthCode
+                                            codeVerifier:codeVerifier
+                                                  appKey:appKey
+                                                  locale:locale
+                                             redirectUri:redirectUri];
+  }
+  return self;
+}
+
+- (void)startWithCompletion:(DBOAuthCompletion)completion {
+  [self startWithCompletion:completion queue:dispatch_get_main_queue()];
+}
+
+- (void)startWithCompletion:(DBOAuthCompletion)completion
+                      queue:(dispatch_queue_t)queue {
+  _retainSelf = self;
```

Comment:
retainSelf is probably unnecessary. I think you can achieve the same behavior with a local variable:

typedef void(^DBOAuthCompletion)(DBOAuthResult *_Nullable);
```
typeof(self) strongSelf = self;
_completion = ^(DBOAuthResult *_Nullable result) { 
    completion(result);
    strongSelf = nil;
}
```